terraform/eval_context_builtin.go

```diff
@@ -14,6 +14,8 @@ import (
 type BuiltinEvalContext struct {
 	PathValue           []string
 	Interpolater        *Interpolater
+	InterpolaterVars    map[string]map[string]string
```

Comment:
I'd love a quick high level explanation for the Interp vars stuff. I can see the pieces but it'd be helpful to hear the overall story.